src/Rules/Deprecations/TypeHintDeprecatedInFunctionSignatureRule.php

```diff
@@ -0,0 +1,106 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Rules\Deprecations;
+
+use PhpParser\Node;
+use PHPStan\Analyser\Scope;
+use PHPStan\Broker\Broker;
+
+class TypeHintDeprecatedInFunctionSignatureRule implements \PHPStan\Rules\Rule
+{
+
+	/** @var Broker */
+	private $broker;
+
+	public function __construct(Broker $broker)
+	{
+		$this->broker = $broker;
+	}
+
+	public function getNodeType(): string
+	{
+		return Node\FunctionLike::class;
```

Comment:
Please rewrite this rule to check only methods and to use `InClassMethodNode`. You can get the method reflection using `$scope->getFunction()`. Once we're satisfied with it, I'll add `InFunctionNode` to have the same rule for functions. This way you can take advantage of the existing PHPStan type resolving and not reinvent it again.

Reply:
This is indeed a much better approach :+1: